libyul/AssemblyStack.h

```diff
@@ -77,8 +78,9 @@ class AssemblyStack
 		m_errorReporter(m_errors)
 	{}
 
-	/// @returns the scanner used during parsing
-	langutil::Scanner const& scanner() const;
+	/// @returns the char stream used during parsing
+	/// @param _sourceName ignored (present for compatibility with CharStreamBySourceName)
+	langutil::CharStream const& charStream(std::string const& _sourceName = {}) const override;
```

Comment:
Wouldn't it be better to return the char stream only if `_sourceName` matches the name in the stream? Or at least assert that they are the same.